### ISSUE TITLE
Add Enumerable#take

### ISSUE THREAD

**Issue by watzon** [crystal-lang/crystal#5761]
After having a short conversation with @bew on Gitter I decided to open this as an issue. Enumerable currently has `first` and `last` methods which return the first and last elements of an Enumerable respectively. They also take an optional `count` parameter which allows you to get the first or last n elements.

I am proposing that `Enumerable#take` be added to compliment `skip`. It should return every element up to the count index.

An example of how this could be done:

```crystal
module Enumerable(T)
  def take(count : Int)
    raise ArgumentError.new("Attempt to take negative size") if count < 0

    array = Array(T).new
    each_with_index do |e, i|
      array << e if i < count
    end
    array
  end
end
```

**Comment by jhass** [comment]
How is it different from `first`?

**Comment by watzon** [comment]
@jhass basically the only difference is that `take` requires an argument. Ruby also has first and take methods that you can use in the same way.

**Comment by Sija** [comment]
@watzon Wrong, Ruby doesn't have `#first(n : Int)` overload, so perhaps that's the reason of existence of another method which **does take** argument.

**Comment by z64** [comment]
Yes, this overload already exists for `first`.

https://github.com/crystal-lang/crystal/blob/80a975cf0a98a4ce8464d0069145131667becd70/src/enumerable.cr#L1245-L1254

**Comment by watzon** [comment]
@z64 it's more about the naming than anything. When I see first I think first element, it's not until I see the API that I think first _n_ elements. Take however makes sense. I want to take 4 elements from the array. It would be a better named alias really.

**Comment by z64** [comment]
That's a fine point for discussion, but we were just confused because the functionality was already there. It wasn't clear if you just didn't know this or wanted something different. You should update the issue description to reflect this.

**Comment by watzon** [comment]
@z64 how's that?

**Comment by jhass** [comment]
We have a pretty strict policy of no aliases in the standard library in particular and no two ways to do something in general, mainly to increase code understandabilty and decrease debate in the community about the right way to do something. 

So to do this we would need to remove `first` with an argument

**Comment by j8r** [comment]
If `take` is more or less like `first`, most will agree there is no point to implement it.

But, if we can merge `first` and `last` in to `take`, why not?

My idea is to add support to negative counts, that will be used to return last *n* elements.

e.g: `enumerable.take(-3)` wil take the last 3 elements, and `enumerable.take(3)` the first ones.

`Enumerable#skip` could also support this.

**Comment by RX14** [comment]
`first(5)` reads fine to me, `take` is actually a little confusing since to me the act of taking means to move them, which implies `take(5) == shift(5)`. I vote down for this: it's an unnecessary breaking change for pretty much negative benefit as far as i'm concerned.

**Comment by watzon** [comment]
@RX14 it wouldn't have to be breaking, just an addition, but whatever works
